### PR TITLE
Implement -f for build_filename

### PR DESCRIPTION
**Pull request by philipcraig**
Implement `-f FILENAME` for alternative to the default `build.ninja` filename

### CODE REVIEW

**Comment by philipcraig** [comment]
closing until regen support added

**Comment by philipcraig** [comment]
regen supported now

File: src/run.rs [hunk 1]
```diff
@@ -23,10 +23,11 @@ enum BuildResult {
 fn build(
     progress: &mut ConsoleProgress,
     parallelism: usize,
+    build_filename: &str,
     regen: bool,
     target_names: &[String],
 ) -> anyhow::Result<BuildResult> {
-    let mut state = trace::scope("load::read", load::read)?;
+    let mut state = trace::scope_str("load::read", load::read, build_filename)?;
```
Comment (evmar) [inline src/run.rs:30]: This can be
`trace::scope("load::read", || load::read(build_filename))?;`

File: src/run.rs [hunk 2]
```diff
@@ -119,11 +128,9 @@ fn run_impl() -> anyhow::Result<i32> {
         return Ok(1);
     }
 
-    if fake_ninja_compat {
-        if matches.opt_present("version") {
-            println!("1.10.2");
-            return Ok(0);
-        }
+    if fake_ninja_compat && matches.opt_present("version") {
```
Comment (evmar) [inline src/run.rs:131]: This seems unrelated

File: src/run.rs [hunk 3]
```diff
@@ -98,8 +99,16 @@ fn run_impl() -> anyhow::Result<i32> {
     // difference matters too much.
     let mut parallelism = usize::from(std::thread::available_parallelism()?);
 
+    let mut build_filename = "build.ninja".to_string();
```
Comment (evmar) [inline src/run.rs:102]: Maybe create this down by where it's used?

File: src/run.rs (hunk 2 above)
Comment (philipcraig) [reply src/run.rs:131]: > This seems unrelated

yes, it's a clippy fix/suggestion. Would you like it removed from this PR?

File: src/run.rs (hunk 1 above)
Comment (philipcraig) [reply src/run.rs]: done

File: src/run.rs (hunk 3 above)
Comment (philipcraig) [reply src/run.rs]: done

File: src/run.rs (hunk 2 above)
Comment (philipcraig) [reply src/run.rs]: i've removed it

**Comment by philipcraig** [comment]
The Windows check is failing as Windows on main is broken. See https://github.com/evmar/n2/issues/29

**Review by evmar — APPROVED** [review]

**Comment by evmar** [comment]
Thanks!  Sorry about the brokenness.
